Add automatic time uniforms to PP shaders

TheIndiv opened the pull request:
Basically what the title says. Quick and simple implementation.

RicardoLuis0 commented on src/common/rendering/hwrenderer/postprocessing/hw_postprocess.cpp:
```diff
@@ -1099,6 +1105,19 @@ void PPCustomShaderInstance::SetUniforms(PPRenderState *renderstate)
 		}
 	}
 
+	auto timeDeltaOffset = FieldOffset.find("InputTimeDelta");
+	if (timeDeltaOffset != FieldOffset.end())
+	{
+		uint8_t *dst = &uniforms[timeDeltaOffset->second];
+		memcpy(dst, &renderstate->TimeDelta, sizeof(float));
+	}
+	auto timeOffset = FieldOffset.find("InputTime");
+	if (timeOffset != FieldOffset.end())
+	{
+		uint8_t *dst = &uniforms[timeOffset->second];
+		memcpy(dst, &renderstate->Time, sizeof(float));
+	}
+
```
just a nitpick, but the uniform struct is for custom uniforms, should this be added as a uniform field instead? otherwise mods near the size limit of uniform structs may stop working since it was increased in size

TheIndiv replied:
Made a push! Do take a look

RicardoLuis0 replied:
the issue isn't UniformStructSize, it's the raw size of the uniform buffer used for the custom uniforms, that buffer is best left only for said custom uniforms, and the input time as just a regular `uniform float InputTime`

RicardoLuis0 replied:
(like, this isn't as much of a problem for opengl, where the minimum guaranteed size is 16kb, but for vulkan, the post-process uniform struct uses push constants, and those have a minimum guaranteed size of 128 **bytes**)

TheIndiv replied:
Made another push, redoing everything from basically scratch. Do take a gander

RicardoLuis0 replied:
looks good yeah 👍
(i'll hold off on approving/merging this until the other post-process PR is in though, since it'll most likely have a merge conflict)

TheIndiv committed:
Add automatic time uniforms to PP shaders.
InputTimeDelta, InputTime, InputTimeGame.
TimeGame pauses when the game pauses.

RicardoLuis0 commented:
re-stating this here:
> looks good yeah 👍
> (i'll hold off on approving/merging this until the other post-process PR is in though, since it'll most likely have a merge conflict)
